pandas/core/ops.py

```diff
@@ -1228,8 +1228,8 @@ def wrapper(left, right):
                             "{op}".format(typ=type(left).__name__, op=str_rep))
 
         elif (is_extension_array_dtype(left) or
-                is_extension_array_dtype(right)):
-            # TODO: should this include `not is_scalar(right)`?
+                (is_extension_array_dtype(right) and not is_scalar(right))):
```

Comment:
Is checking that it is a string a bit more logical?

Reply:
I guess that would work.  Elsewhere the same check is done with `not is_scalar`

Reply:
then ignore my comment and go for consistency